applications/TopologyOptimizationApplication/custom_elements/comparison_utilities.h

```diff
@@ -0,0 +1,342 @@
+//
+//   Project Name:        KratosSolidMechanicsApplication $
+//   Created by:          $Author:            JMCarbonell $
+//   Last modified by:    $Co-Author:                     $
+//   Date:                $Date:                July 2013 $
+//   Revision:            $Revision:                  0.0 $
+//
+//
+
+#if !defined(KRATOS_COMPARISON_UTILITIES )
+#define  KRATOS_COMPARISON_UTILITIES
+
+
+/* System includes */
+
+/* Project includes */
+#include "math_utilities.h"
+
+
+namespace Kratos
+{
+
+/**@name Kratos Globals */
+/*@{ */
+
+
+/*@} */
+/**@name Type Definitions */
+/*@{ */
+
+/*@} */
+
+
+/**@name  Enum's */
+/*@{ */
+
+
+/*@} */
+/**@name  Functions */
+/*@{ */
+
+
+
+/*@} */
+/**@name Kratos Classes */
+/*@{ */
+
+
+class ComparisonUtilities
+{
+public:
+    /**@name Type Definitions */
+    /*@{ */
+
+
+    /*@} */
+    /**@name Life Cycle
+     */
+    /*@{ */
+
+    /** Constructor.
+     */
+    ComparisonUtilities() {};
+
+    /** Destructor.
+     */
+    ~ComparisonUtilities() {};
+
+    /** Operators.
+     */
+
+    //**************************************************************************
+    //**************************************************************************
+
+
+    //**************************************************************************
+    //**************************************************************************
+
+    double CalculateStressNorm(const Vector& StressVector)
```

Comment:
If no template, move the code to a cpp file

Reply:
Would you reccomend to move this code part to the `small_displacement_simp_element.cpp`? Or am I getting it wrong to move it to a cpp file?

Reply:
No, what I recommend is not to have all code only in header. Each time a header is included, the whole code is "copy-pasted", making the compilation much slower. Summary: Create comparison_utilities.cpp and move the code there